src/cc/bcc_syms.cc

```diff
@@ -0,0 +1,258 @@
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <unordered_map>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "bcc_helpers.h"
+
+class SymbolCache {
+public:
+	virtual void refresh() = 0;
+	virtual bool resolve_addr(uint64_t addr, struct bcc_symbol *sym) = 0;
+	virtual bool resolve_name(const char *name, uint64_t *addr) = 0;
+};
+
+class KSyms : SymbolCache {
+	struct Symbol {
+		Symbol(const char *name, uint64_t addr) :
+			name(name), addr(addr) {}
+		std::string name;
+		uint64_t addr;
+
+		bool operator<(const Symbol& rhs) const { return addr < rhs.addr; }
+	};
+
+	std::vector<Symbol> _syms;
```

Comment:
> When using clang-format command line utility or clang::format::reformat(...) functions from code, one can either use one of the predefined styles (LLVM, **Google**, Chromium, Mozilla, WebKit) or create a custom style by configuring specific style options.

It seems like the ideal choice here. Particularly since this whole project is already based on LLVM.